cristin-organization/src/test/java/no/unit/nva/cristin/organization/query/QueryCristinOrganizationHandlerTest.java

```diff
@@ -252,6 +261,55 @@ void shouldHaveCorrectCristinUriWithParamsOnVersionTwo() throws Exception {
         assertThat(gatewayResponse.getStatusCode(), equalTo(HTTP_OK));
     }
 
+    @Test
+    void shouldHaveFullTreeOfOrganizationsWhenRequested() throws Exception {
+        var fetchClient = mockFetchClient();
+        queryCristinOrgClient20230526 = new QueryCristinOrgClient20230526(httpClient, fetchClient);
+        queryCristinOrgClient20230526 = spy(queryCristinOrgClient20230526);
+
+        var fakeQueryResponseResource = IoUtils.stringFromResources(Path.of(CRISTIN_QUERY_RESPONSE_V_2_JSON));
+        doReturn(new HttpResponseFaker(fakeQueryResponseResource))
+            .when(queryCristinOrgClient20230526).fetchQueryResults(any());
+
+        doReturn(queryCristinOrgClient20230526).when(clientProvider).getVersion20230526();
+
+        queryCristinOrganizationHandler = new QueryCristinOrganizationHandler(clientProvider, new Environment());
+        var input = handlerRequestWantingFullTree();
+        queryCristinOrganizationHandler.handleRequest(input, output, context);
+
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, SearchResponse.class);
+        var responseBody = gatewayResponse.getBodyObject(SearchResponse.class);
+        var actualHits = convertHitsToProperFormat(responseBody);
+
+        assertThat(gatewayResponse.getStatusCode(), equalTo(HTTP_OK));
+        assertThat(actualHits.get(0).getHasPart().size(), equalTo(8));
+        assertThat(actualHits.get(1).getHasPart().size(), equalTo(8));
+    }
+
+    @Test
+    void shouldDropEnrichmentWhenIdentifierNotFoundInUpstream() throws Exception {
+        var fetchClient = mockFetchClientWithOneHitMissingInUpstream();
+        queryCristinOrgClient20230526 = new QueryCristinOrgClient20230526(httpClient, fetchClient);
+        queryCristinOrgClient20230526 = spy(queryCristinOrgClient20230526);
+
+        var fakeQueryResponseResource = IoUtils.stringFromResources(Path.of(CRISTIN_QUERY_RESPONSE_V_2_JSON));
+        doReturn(new HttpResponseFaker(fakeQueryResponseResource))
+            .when(queryCristinOrgClient20230526).fetchQueryResults(any());
+
+        doReturn(queryCristinOrgClient20230526).when(clientProvider).getVersion20230526();
+
+        queryCristinOrganizationHandler = new QueryCristinOrganizationHandler(clientProvider, new Environment());
+        var input = handlerRequestWantingFullTree();
+        queryCristinOrganizationHandler.handleRequest(input, output, context);
+
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, SearchResponse.class);
+        var responseBody = gatewayResponse.getBodyObject(SearchResponse.class);
+        var actualHits = convertHitsToProperFormat(responseBody);
+
+        assertThat(gatewayResponse.getStatusCode(), equalTo(HTTP_OK));
+        assertThat(actualHits.size(), equalTo(1));
+    }
```

Comment:
Is there a test for e.g. gateway exception?

Reply:
Done